src/server.ts

```diff
@@ -1,58 +1,45 @@
-import express from 'express';
-import helmet from 'helmet';
-import { json } from 'body-parser';
-import cors from 'cors';
-import passport from 'passport';
-import { Server } from 'http';
+import { ENV } from '@/utils/env';
+import { logger, LOG_LEVEL } from './logger';
+
+export const start = async () => {
+  logger.info(`Log level: ${LOG_LEVEL}`);
+
+  /**
+   * Async imports allow to dynamically import the required modules,
+   * and therefore only import modules that are required, when they are required.
+   * It decreases the loading time on startup.
+   * If we decide to apply migrations/metadata in a separate moment e.g. k8s initContainer,
+   * it will allow hasura-auth to load modules only required for migrations/metadata,
+   * or not to load these modules when auto migration/metadata is disabled.
```

Comment:
This feels Nhost-specific and should probably not be in the Hasura Auth source code. Maybe we can store this information somewhere else?

Reply:
I rephrased the comment so it does not imply it is related to our infra, but in the way hasura-auth can be generally orchestrated